invoices/interface.go

```diff
@@ -18,4 +18,8 @@ type Payload interface {
 	// CustomRecords returns the custom tlv type records that were parsed
 	// from the payload.
 	CustomRecords() record.CustomSet
+
+	// Metadata returns the additional data that is sent along with the
+	// payment to the payee.
+	Metadata() []byte
```

Comment:
One thing I'm missing here, don't we want too validate that we have metadata present when the payload belongs to an invoice that had metadata? From spec, looks like we need to return invalid details err. Or is that already implemented somewhere and I've missed it.

Reply:
I think this related to https://github.com/lightningnetwork/lnd/pull/5810#discussion_r782100591

The idea is that this prepares for never having any invoice data (also not the presence of metadata itself) available.